### PR TITLE
Feat/1520 qualification accordion

### PR DESCRIPTION
**Pull request by kapppa-joe**
#### Work done
- Add a select-qualification-type component
- Store the selected qualification at QualificationService
- Patch a minor bug at GroupedRadioButtonAccordionComponent

#### Tests
Does this PR include tests for the changes introduced?
- [x] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: frontend/src/app/features/training-and-qualifications/add-edit-qualification/select-qualification-type/select-qualification-type.component.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+  constructor(
+    protected formBuilder: FormBuilder,
+    protected qualificationService: QualificationService,
+    protected router: Router,
+    protected backLinkService: BackLinkService,
+    protected workerService: WorkerService,
+    protected route: ActivatedRoute,
+    protected errorSummaryService: ErrorSummaryService,
+  ) {}
```
Comment (duncanc19) [inline frontend/src/app/features/training-and-qualifications/add-edit-qualification/select-qualification-type/select-qualification-type.component.ts:62]: I think it's a bit unusual to have the ngOnInit above the constructor, or at least I don't think we do it in this codebase much!

Comment (duncanc19) [reply frontend/src/app/features/training-and-qualifications/add-edit-qualification/select-qualification-type/select-qualification-type.component.ts:62]: Also, I think these classes can be private, protected is usually used for either directives or components which extend a directive

File: frontend/src/app/features/training-and-qualifications/add-edit-qualification/select-qualification-type/select-qualification-type.component.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+  protected setTitle(): void {
+    this.title = 'Select the type of qualification you want to add';
+  }
```
Comment (duncanc19) [inline frontend/src/app/features/training-and-qualifications/add-edit-qualification/select-qualification-type/select-qualification-type.component.ts:66]: With a few of these setter functions, I'm wondering if we get much benefit from them as they're not dynamic, the text could just go directly in the html template

**Review by Jonopono123 — APPROVED** [review]

File: frontend/src/app/features/training-and-qualifications/add-edit-qualification/select-qualification-type/select-qualification-type.component.ts (hunk 1 above)
Comment (kapppa-joe) [reply frontend/src/app/features/training-and-qualifications/add-edit-qualification/select-qualification-type/select-qualification-type.component.ts]: Thanks!  Not sure why I put it like that. Now ngOnInit is below the constructor.
Also changed the protected vars to private

**Review by duncanc19 — APPROVED** [review]

**Pull request merged**